src/containers/BookForm.js

```diff
@@ -1,19 +1,79 @@
-import React from 'react';
-
-const BookForm = () => {
-  const categories = ['Action', 'Biography', 'History', 'Horror', 'Kids', 'Learning', 'Sci-Fi'];
-  return (
-    <form>
-      <label htmlFor="title">
-        Title
-        <input id="title" type="text" name="title" />
-      </label>
-      <select id="category">
-        {categories.map(e => <option key={e} value={e}>{e}</option>)}
-      </select>
-      <button type="submit">Submit</button>
-    </form>
-  );
+import React, { Component } from 'react';
+import { connect } from 'react-redux';
+import PropTypes from 'prop-types';
+import { createBook } from '../actions/index';
+
+const categories = [
+  'Action',
+  'Biography',
+  'History',
+  'Horror',
+  'Kids',
+  'Learning',
+  'Sci-Fi',
+];
+
+class BookForm extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      id: Math.floor(Math.random() * 100),
+      title: '',
+      category: categories[0],
+    };
+
+    this.handleChange = this.handleChange.bind(this);
+    this.handleSubmit = this.handleSubmit.bind(this);
+  }
+
+  async handleChange(e) {
+    const { name } = e.target;
+    this.setState({
+      [name]: e.target.value,
+    });
+  }
```

Comment:
- [ ] OPTIONAL - No use making this function async. Consider making it a regular function. Async usually goes with await and so far in this there is n o async event to wait for a return value from it.